### PR TITLE
Add management CLI sample

### PR DESCRIPTION
**Pull request by shilgapira**
## Description
Adds a code sample that builds a CLI utility for managing users and tenants.

## Usage
```shell
cd examples/managementcli
go build .
export DESCOPE_PROJECT_ID=...
export DESCOPE_MANAGEMENT_KEY=...
./managementcli help
```

## Must
- [X] Tests
- [X] Documentation (if applicable)

cli

### CODE REVIEW

File: examples/managementcli/managementcli.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	Tenants    []string
+	Domains    []string
+}
+
+// Descope SDK
+
+var descopeClient *descope.DescopeClient
+
+func prepare() (err error) {
+	if utils.GetProjectIDEnvVariable() == "" {
```
Comment (dorsha) [inline examples/managementcli/managementcli.go:30]: Maybe we can ask for it.. and if provided save to env file (and put it in gitignore)

File: examples/managementcli/managementcli.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+		cmd.Args = cobra.ExactArgs(2)
+		cmd.Flags().StringSliceVarP(&flags.Domains, "domains", "D", nil, "the tenant's self provisioning domains")
+	})
+
+	addCommand(tenantDelete, "tenant-delete <id>", "Delete an existing tenant", func(cmd *cobra.Command) {
+		cmd.Args = cobra.ExactArgs(1)
+		cmd.DisableFlagsInUseLine = true
+	})
+
+	err := cli.Execute()
```
Comment (dorsha) [inline examples/managementcli/managementcli.go:148]: Do we log the error?

**Review by dorsha — COMMENTED** [review]
Very nice!! small comment.

Maybe we can have a test that verify the cli commands match the actual proto rpc, so we won't forget to add ones at the future, and it will be easier to maintain.

And consider adding tests to this cli, not a must but nice to have. Simple ones.

File: examples/managementcli/managementcli.go (hunk 1 above)
Comment (dorsha) [reply examples/managementcli/managementcli.go]: 👍

**Comment by dorsha** [comment]
Looks good, check my comment regarding coverage and tests.

File: .gitignore [hunk 3]
```diff
@@ -16,6 +16,7 @@
 *.dylib
 examples/webapp/webapp
 examples/ginwebapp/ginwebapp
+examples/managementcli/managementcli
```
Comment (asafshen) [inline .gitignore:19]: @Bars92 do devs need to add this to go.work after this PR is merged? (I see I have the other examples  it in comment)

**Review by dorsha — DISMISSED** [review]

**Review by dorsha — APPROVED** [review]

**Pull request merged**